### PR TITLE
Fix a name of an internal _onRequestWithOptions property

### LINKED ISSUE #1480: TypeError: _onRequestWithOpts is not a function
### [REQUIRED] Environment info

**firebase-tools:** 7.0.2

**Platform:** macOS

### [REQUIRED] Test case

`npm i -g firebase-tools@latest` or
`npm i -D firebase-tools@latest`

### [REQUIRED] Steps to reproduce

Error encountered: TypeError: _onRequestWithOpts is not a function at Object.httpsProvider._onRequestWithOpts (in lib/emulator/functionsEmulatorRuntime.js)

1) Init a new project with Firebase init, choose location, and set up functions in the dashboard
2) Updated global package, ran `firebase run serve`, and encountered the error
3) Deleted the global package, installed locally to project, ran `firebase run serve`, and encountered the error
4) Console logged `const httpsProvider = require(httpsProviderResolution);` in the above file and the path outputted is `lib/providers/https` which does not exist.

(Note: Functions local emulator worked prior to step #2 when I tried to update firebase-tools.)

### [REQUIRED] Expected behavior

Run functions locally.

### [REQUIRED] Actual behavior

functions: Emulator started at http://localhost:5000
i  functions: Watching "/Users/joe/Server/my-project/functions" for Cloud Functions...
⚠  TypeError: _onRequestWithOpts is not a function
    at Object.httpsProvider._onRequestWithOpts (/Users/joe/Server/my-project/functions/node_modules/firebase-tools/lib/emulator/functionsEmulatorRuntime.js:278:24)
    at Object.httpsProvider.onRequest (/Users/joe/Server/my-project/functions/node_modules/firebase-tools/lib/emulator/functionsEmulatorRuntime.js:283:34)
    at Object.<anonymous> (/Users/joe/Server/my-project/functions/index.js:18:38)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at /Users/joe/Server/my-project/functions/node_modules/firebase-tools/lib/emulator/functionsEmulatorRuntime.js:582:33
⚠  We were unable to load your functions code. (see above)

### PR DESCRIPTION
**Pull request by merlinnot**
### Description

Resolves #1480 

Apparently this repository relies on a private method defined in `firebase-functions`, which was renamed as part of https://github.com/firebase/firebase-functions/pull/495. This PR updates the name of the method.
	 
### Scenarios Tested

Unfortunately I was not able to run tests locally by simply doing `npm install && npm run test`, even on the master branch. I'll keep trying, but I thought that this PR might be simple enough to create it already, without 

### Sample Commands

Not relevant.

### CODE REVIEW

**Comment by merlinnot** [comment]
CI is green, so I guess issues with tests are just on my local machine.

**Comment by merlinnot** [comment]
@thechenky for a review

**Comment by samtstern** [comment]
@merlinnot I was able to reproduce the bug on master and then make it go away by applying this PR.

**Comment by samtstern** [comment]
@merlinnot the problem with this is now the opposite.  If you're using `firebase-functions@3.0.x` then emulation will fail because this code now relies on internal APIs of `3.1.0`.

So we either need to:
a) Increase the minimum version of the functions SDK for emulation to `3.1.0` (which will require changing some logic as we only check majors right now).
b) Introduce logic to branch on the version and choose the right internal method name.

And in the long term we need to stop relying on internal APIs at all, but let's get a quick fix in first.

I will send you a PR that does option `b`

**Comment by samtstern** [comment]
@merlinnot see https://github.com/merlinnot/firebase-tools/pull/1

**Comment by merlinnot** [comment]
Sure, let's do that.

File: changelog.txt [hunk 1]
```diff
@@ -6,3 +6,4 @@
 * Hot reload firestore rules on change.
 * Upgrade archiver dependency to 3.0.0.
 * Update functions init templates to `v3.1.0`
+* Fix emulation of https functions
```
Comment (bkendall) [inline changelog.txt]: Put a period on the end of the sentence.

Comment (merlinnot) [reply changelog.txt]: 07d718e28f280c2c4ef5103a761ff9da73101a14

**Comment by samtstern** [comment]
Thank you @merlinnot!

**Comment by Primology** [comment]
I was able to fix the issue by downgrading to **firebase-functions@3.0.2** , otherwise while using **firebase-functions@3.1.0** I still have the following issue -> **TypeError: _onRequestWithOpts is not a function**

**Comment by isabelleyzhou** [comment]
if I have the same issue, is downgrading to 3.0.2 the only thing I need to do to receive these changes? because i'm still getting the same error despite the downgrade.

**Comment by tcha-tcho** [comment]
@isabelleyzhou I think I found our problem. The version of firebase-tools installed by Google is outdated on global envy. Try to run `sudo npm i -g firebase-tools` (you will need the version 7.2.4). After that don't use `firebase serve --only functions` it won't work, you will need to use: `firebase emulators:start --only functions`.

**Pull request merged**